File: sea-shell-service-immutable/src/main/java/org/jesperancinha/shell/webflux/immutable/data/SeaShellCostumeDto.java
package org.jesperancinha.shell.webflux.immutable.data;

import lombok.Builder;

public record SeaShellCostumeDto(
        SeaShellTopDto topDto,
        SeaShellLowerDto lowerDto,
        Long topId,
        Long lowerId
) {
    @Builder
    public SeaShellCostumeDto {
    }
}

File: sea-shell-service-immutable/src/main/kotlin/org/jesperancinha/shell/webflux/immutable/data/SeaShellCostumeDto.kt
package org.jesperancinha.shell.webflux.immutable.data

import lombok.Builder

@JvmRecord
data class SeaShellCostumeDto @Builder constructor(
    val topDto: SeaShellTopDto,
    val lowerDto: SeaShellLowerDto,
    val topId: Long,
    val lowerId: Long
)